src/main/java/org/hl7/davinci/priorauth/ClaimEndpoint.java

```diff
@@ -599,7 +606,36 @@ private Bundle generateAndStoreClaimResponse(Bundle bundle, Claim claim, String
 
     @Override
     public void run() {
-      updateClaim(bundle, claimId, patient, disposition);
+      if (updateClaim(bundle, claimId, patient, disposition) != null) {
+        // Check for subscription
+        Map<String, Object> constraintMap = new HashMap<String, Object>();
+        constraintMap.put("claimResponseId", claimId);
+        constraintMap.put("patient", patient);
+        Subscription subscription = (Subscription) App.getDB().read(Database.SUBSCRIPTION, constraintMap);
```

Comment:
could there be more than one?

Reply:
I could have sworn I put a comment on here about that - I wrote it down somewhere...
The IG does not specify but yes this implementation only supports one Subscription per Claim

Reply:
What happens if a second one comes in though, do you handle it by replacing the entry in the database, or do you just keep them all as a record and respond to the latest one or first one that came in?